include/bbp/sonata/report_reader.h

```diff
@@ -23,7 +23,8 @@ struct SONATA_API DataFrame {
     std::vector<double> times;
     DataType ids;
     // data[times][ids]
-    std::vector<std::vector<float>> data;
+    size_t n_cols, n_rows;
+    std::vector<float> data;
```

Comment:
Do we have to live without other libraries? What about `boost::numeric::ublas::matrix`? It offers a simple API and ensures memory contiguity.

Reply:
I think this is sufficient, don't see the point in adding the heavy-handed Boost dependency. But as you said, one of rows/cols could be inferred all the time. As this is contained internally, I don't see it as a big issue, though.

Reply:
(If there is no existing boost dependency, I would definitely avoid it)